### PR TITLE
Issue/13743 display total discount in checkout

### LINKED ISSUE #13743: [Woo POS]: Display discounted amount in price summary when coupon applied – Dummy UI
One or more coupons might be applied to the order, we might want to display the total discounted amount on the right hand side of the checkout screen, somewhere above totals and taxes.

### PR DESCRIPTION
**Pull request by malinajirka**
Closes: #13743 

### Description

Display total discount on the checkout summary screen in the POS. The discount is shown only when Coupons feature flag is enabled.

The position of the discount row is random at the moment and doesn't contain `-` character since implementation of these will depend on the final designs from a designer.

### Testing information

- Verify the total discount is displayed when the discount is non-zero.
- Verify the total discount is not displayed when the discount is zero.
- Verify the total discount is not displayed when FF is disabled.

### The tests that have been performed

I've performed the above tests. I haven't tested any edge cases since currently the coupons feature supports only the most happy path with dummy implementation of coupon selection.

### Images/gif

Screenshot_1743084689

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsScreen.kt [hunk 1]
```diff
@@ -324,6 +325,15 @@ private fun TotalsGrid(totals: Totals.Visible) {
         horizontalAlignment = Alignment.CenterHorizontally,
         verticalArrangement = Arrangement.Center,
     ) {
+        if (FeatureFlag.POS_COUPONS.isEnabled() && totals.orderDiscountText != null) {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsScreen.kt]: Probably makes sense to move this logic to the VM. And use there instead of the static FF, wrapper which will allow to write a test if needed.

Here it can be then just 

```
        totals.orderDiscountText?.let {
            TotalsGridRow(
                textOne = stringResource(R.string.woopos_payment_discount_label),
                textTwo = it,
            )

            Spacer(modifier = Modifier.height(WooPosSpacing.Medium.value.toAdaptivePadding()))            
        }
```

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt [hunk 2]
```diff
@@ -466,6 +467,7 @@ class WooPosTotalsViewModel @Inject constructor(
         }
         return WooPosTotalsViewState.Checkout(
             totals = Totals.Visible(
+                orderDiscountText = if (discountAmount > BigDecimal.ZERO) priceFormat(discountAmount) else null,
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModel.kt]: The FF flag via wrapper can go here

**Review by kidinov — APPROVED** [review]
LGTM! I left one suggestion regarding code structure

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsScreen.kt (hunk 1 above)
Comment (malinajirka) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsScreen.kt]: Fixed in https://github.com/woocommerce/woocommerce-android/pull/13859/commits/b59ffd97d04ce51bf128469fb8f1e50dade74962
